app/Commands/EditEnvironmentVariables.php

```diff
@@ -0,0 +1,174 @@
+        ], $this->helpers->aws()->commonConsoleOptions()));
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $requiredBinaries = ['aws', config('app.editor')];
+
+        if ($this->helpers->checks()->checkAndReportMissingBinaries($this, $requiredBinaries)) {
+            exit(1);
+        }
+
+        $bucket = $this->option('client-system') ?? getenv('AWS_S3_ENV');
+
+        if (!$bucket) {
+            $this->error('An S3 bucket for the environment variable files must be specified.');
+            exit(1);
+        }
+
+        $this->processUpdate($bucket);
+
+        exit(0);
+    }
+
+    private function processUpdate(string $bucket): void
+    {
+        $menuItem = $this->getMenuItem($bucket);
+
+        if ($menuItem === null) {
+            exit(0);
+        }
+
+        $this->getFile($menuItem, $bucket);
+
+        $newContents = $this->editFile();
+
+        $this->checkDiffAndUpload($newContents, $bucket);
+
+        exit(0);
+    }
+
+    private function getMenuItem(string $bucket): ?int
+    {
+        $this->envFiles = collect($this->helpers->aws()->s3()->listFiles($this, $bucket))
+            ->pluck('Key')
+            ->map(function ($filename) {
+                return $filename;
+            })
+            ->reject(function ($filename) {
+                return !Str::endsWith($filename, '.env');
+            })
+            ->toArray();
+
+        $additionalMenuOptions = ['Create new file'];
+        $callerArn = $this->helpers->aws()->iam()->getCallerArn($this);
+        $callerArnParts = explode('/', $callerArn);
+
+        if ($callerArnParts[1] == 'NetsellsSecurityOps') {
+            $additionalMenuOptions[] = 'Delete file';
+        }
+
+        return $this->menu("Choose a file to edit", array_merge($this->envFiles, $additionalMenuOptions))->open();
+    }
+
+    private function getFile(int $menuItem, string $bucket): void
+    {
+        if ($menuItem < count($this->envFiles)) {
+            $this->fileName = $this->envFiles[$menuItem];
+            $this->fileContents = (string)$this->helpers->aws()->s3()->getFile($this, $bucket, $this->fileName)->get('Body');
+        } elseif ($menuItem == count($this->envFiles)) {
+            $this->fileName = $this->ask('Enter name of new file (must have an extension of .env)');
+            if (!Str::endsWith($this->fileName, '.env')) {
+                exit(1);
+            }
+            $this->fileContents = '';
+        } else {
+            $deleteMenuItem = $this->menu("Choose a file to edit",$this->envFiles)->open();
+            if ($deleteMenuItem !== null) {
+                if ($this->confirm("Are you sure you wish to delete {$this->envFiles[$deleteMenuItem]}?")) {
+                    $this->helpers->aws()->s3()->deleteFile($this, $bucket, $this->envFiles[$deleteMenuItem]);
```

Comment:
A confirmation ->info() to let the user know this was a success here would be good

Reply:
Done

Reply:
Done